pg/src/main/java/org/bouncycastle/openpgp/PGPSignature.java

```diff
@@ -599,6 +599,12 @@ public void encode(
         boolean forTransfer)
         throws IOException
     {
+        // Exportable signatures MUST NOT be exported if forTransfer==true
+        if (forTransfer && (!getHashedSubPackets().isExportable() || !getUnhashedSubPackets().isExportable()))
```

Comment:
I am up for debate, whether or not we should respect Exportable subpackets in the unhashed section of the signature.

Reply:
I think, until we have information suggesting we shouldn't, we should. It's easy to add an exception later, it can be very hard to withdraw an exception once it's made though.

Reply:
Yes, agreed. I just thought I'd mention it, in theory unhashed packets can be added and removed at will, but it wouldn't be the first time theory does not quite match with reality. I'd say what's just been added is an improvement on before at least, that's the main thing.

Reply:
I talked to some people at Sequoia-PGP and they think that simply wiping the unhashed area is not a good idea.
Instead they proposed to add a `normalize()` method, which removes subpackets following some well-defined rules: